cpp/include/cudf/detail/unary_detail.hpp

```diff
@@ -0,0 +1,61 @@
+/*
```

Comment:
This file can just be named `unary.hpp`. No need for the `_detail` suffix.

Reply:
It would be easier to distinguish between normal header file and the one within detail namespace